addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -995,6 +1026,32 @@ func ownedByDifferentCluster(k8SObject *unstructured.Unstructured, cluster *clus
 	return ""
 }
 
+func (h *Helper) setLabels(labels map[string]string, child *unstructured.Unstructured) error {
+	gvr, err := h.GVRHelper.GetGVR(child.GroupVersionKind().GroupKind())
+	if err != nil {
+		h.Logger.Error(err, fmt.Sprintf("unable to get GVR of %s/%s", child.GetNamespace(), child.GetName()))
+		return err
+	}
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// We need to get and update, otherwise there could have concurrency issue: ["the object has been modified; please
+		// apply your changes to the latest version and try again"]
+		newChild, errGetProvider := h.DynamicClient.Resource(*gvr).Namespace(child.GetNamespace()).Get(h.Ctx, child.GetName(), metav1.GetOptions{})
+		if errGetProvider != nil {
+			h.Logger.Error(errGetProvider, fmt.Sprintf("unable to get %s %s/%s", child.GetKind(), child.GetNamespace(), child.GetName()))
+			return errGetProvider
+		}
+		newChild = newChild.DeepCopy()
+		newChild.SetLabels(labels)
+		_, errUpdateProvider := h.DynamicClient.Resource(*gvr).Namespace(newChild.GetNamespace()).Update(h.Ctx, newChild, metav1.UpdateOptions{})
```

Comment:
How about use patch to avoid update conflict?

Reply:
> 

Sounds better, although multiple similar methods use the same approach to update the resources in  the same file , maybe better to do a whole refactor later.

Reply:
Will only this place update the labels? If so, patch will simplify and avoid conflict check.

Reply:
Sure, only in this controller. i will update to use patch method

Reply:
updated.